src/cargo/sources/git/oxide.rs

```diff
@@ -0,0 +1,61 @@
+//! This module contains all code sporting `gitoxide` for operations on `git` repositories and it mirrors
+//! `utils` closely for now. One day it can be renamed into `utils` once `git2` isn't required anymore.
+
+use crate::util::{network, Progress};
+use crate::{CargoResult, Config};
+use git_repository as git;
+use std::sync::atomic::AtomicBool;
+use std::sync::Arc;
+use std::time::Duration;
+
+/// For the time being, `repo_path` makes it easy to instantiate a gitoxide repo just for fetching.
+/// In future this may change to be the gitoxide repository itself.
+pub fn with_retry_and_progress(
+    repo_path: &std::path::Path,
+    config: &Config,
+    cb: &mut (dyn FnMut(
+        &git::Repository,
+        &AtomicBool,
+        &mut git::progress::tree::Item,
+    ) -> CargoResult<()>
+              + Send),
+) -> CargoResult<()> {
+    let repo = git::open_opts(repo_path, {
+        let mut opts = git::open::Options::default();
+        // We need `git_binary` configuration as well for being able to see credential helpers
+        // that are configured with the `git` installation itself.
+        // However, this is slow on windows (~150ms) and most people won't need it as they use the
+        // standard index which won't ever need authentication.
+        // TODO: This is certainly something to make configurable, at the very least on windows.
+        //       Maybe it's also something that could be cached, all we need is the path to the configuration file
+        //       which usually doesn't change unless the installation changes. Maybe something keyed by the location of the
+        //       binary along with its fingerprint.
```

Comment:
Forgot to call this out before.  What will be our method for tracking this TODO.  Or if it isn't worth tracking, should be be more honest with our selves and re-frame this as documentation (Frequently, my `TODO`s get re-framed as `HACK`s)

Reply:
This TODO was removed in a later commit and I hope that's OK, particularly because I wanted to avoid rewriting the history this far in the review process with the goal of squashing in the end.